src/components/ApplyView.tsx

```diff
@@ -0,0 +1,297 @@
+      ...change,
+      accepted: true,
+      rejected: false,
+    }));
+  });
+
+  // Group changes into blocks for better UI presentation
+  const [changeBlocks, setChangeBlocks] = useState<ExtendedChange[][]>([]);
+
+  // Process diff into blocks of related changes
+  useEffect(() => {
+    if (!diff.length) return;
+
+    const blocks: ExtendedChange[][] = [];
+    let currentBlock: ExtendedChange[] = [];
+    let inChangeBlock = false;
+
+    diff.forEach((change) => {
+      if (change.added || change.removed) {
+        if (!inChangeBlock) {
+          inChangeBlock = true;
+          currentBlock = [];
+        }
+        currentBlock.push(change);
+      } else {
+        if (inChangeBlock) {
+          blocks.push([...currentBlock]);
+          currentBlock = [];
+          inChangeBlock = false;
+        }
+        blocks.push([change]);
+      }
+    });
+
+    if (currentBlock.length > 0) {
+      blocks.push(currentBlock);
+    }
+
+    setChangeBlocks(blocks);
+  }, [diff]);
+
+  // Add defensive check for state after hooks
+  if (!state || !state.originalContent || !state.newContent) {
+    console.error("Invalid state:", state);
+    return (
+      <div className="flex flex-col h-full items-center justify-center">
+        <div className="text-error">Error: Invalid state - missing content</div>
+        <Button onClick={close} className="mt-4">
+          Close
+        </Button>
+      </div>
+    );
+  }
+
+  // Handle accepting all changes that have been marked as accepted
+  const handleAccept = async () => {
+    try {
+      // Filter out rejected changes and include both accepted and unselected changes
+      const newContent = diff
+        .filter((change) => {
+          // Handle rejected changes
+          if (change.rejected) {
+            if (change.added) return false;
+            else if (change.removed) return true;
+          }
+          // Handle accepted or unselected changes
+          if (change.removed) return false;
+          return true;
+        })
+        .map((change) => change.value)
+        .join("");
+
+      await app.vault.modify(state.file, newContent);
+      new Notice("Changes applied successfully");
+      close();
+    } catch (error) {
+      console.error("Error applying changes:", error);
+      new Notice(`Error applying changes: ${error.message}`);
+    }
+  };
+
+  // Handle rejecting all changes
+  const handleReject = () => {
+    close();
+  };
+
+  // Accept a block of changes
+  const acceptBlock = (blockIndex: number) => {
+    setDiff((prevDiff) => {
+      const newDiff = [...prevDiff];
+      const block = changeBlocks[blockIndex];
+
+      // Find the indices of the changes in this block
+      block.forEach((blockChange) => {
+        const index = newDiff.findIndex((change) => change === blockChange);
+        if (index !== -1) {
+          newDiff[index] = {
+            ...newDiff[index],
+            accepted: true,
+            rejected: false,
```

Comment:
why do we need two flags to represent this value? is there a case where accepted and rejected can be same values?